activerecord/lib/active_record/relation.rb

```diff
@@ -255,6 +255,8 @@ def find_or_create_by!(attributes, &block)
     #   the problem of running out of integers, if the underlying table is still stuck on a primary
     #   key of type int (note: All \Rails apps since 5.1+ have defaulted to bigint, which is not liable
     #   to this problem).
+    # * Columns with unique database constraints should not have uniqueness validations defined,
+    #   otherwise #create will fail with ActiveRecord::RecordInvalid and #find_by will never be called.
```

Comment:
Calling `#find_or_create_by` without a `!` will not raise an `ActiveRecord::RecordInvalid` but fail silently. We can copy some wording from line 263 which makes it applicable to the `!` version as well:
```suggestion
    #   otherwise #create will fail due to validation errors and #find_by will never be called.
```